### DISCUSSION TITLE
How to complete disable SSR?

### DISCUSSION THREAD

**Discussion by realtebo** [vercel/next.js#73355] [Help]
### Summary

I'm having issues on amplify due to SSR.
I have almost all app using 'use client';
Anyway, some SSR still is occouring and it's breaking deployment
Can I temporary disable SSR at all!?!?

**Comment by icyJoseph** [comment]
`use client` doesn't disable SSR, https://github.com/reactwg/server-components/discussions/4

To do that, there's a few tricks:

- https://demystifying-rsc.vercel.app/client-components/next-dynamic/
- https://demystifying-rsc.vercel.app/client-components/no-ssr/

**Reply by icyJoseph** [reply]
With that being said, what problems exactly are you running into?

**Comment by realtebo** [comment]
Thanks for replying.

I received a design system, from a 3rd party, which Is using `window` function in the code. So while Building It Is breaking the build because this function doesn't exists server side. It serma that adding use client here and there Is not enough.

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by icyJoseph** [reply]
If the window is accessed without guards in the component function body, then deferred state reveals are the way to go:

- https://demystifying-rsc.vercel.app/client-components/no-ssr/

There's no like, blanket flag or w.e. to opt out, AFAIK

Lazy loading would also work, but typically that's more for cases where the window is accessed also at top of the module level.
